### DISCUSSION TITLE
How to turn off retry mutation on device re-connect

### DISCUSSION THREAD

**Discussion by nguyenyou** [TanStack/query#3713] [Q&A] [answered]
```
If mutations fail because the device is offline, they will be retried in the same order when the device reconnects.
```
In case this is not the behavior that we want, is there any way to turn this off? Please help.

**Comment by TkDodo** [comment]
What would you want to happen instead?

**Reply by nguyenyou** [reply]
I want it not to do anything.

**Comment by TkDodo** [comment, marked as answer]
In v4, you can set `networkMode: 'always'` to not pause retries.

**Reply by nguyenyou** [reply]
In our case, the user clicked on a pay-now button, because their internet connection is not stable, they click multiple times on it. After that, their internet connection is reconnected, consequently they made multiple transactions because of those **failed requests** got retried.

I want those requests not to be retried at all, let the user do it themself.

**Reply by TkDodo** [reply]
1) I would disable the button after the first click
2) mutations per default don't have retries, so if the user should retry manually - don't use the retry option

**Reply by nguyenyou** [reply]
1. The request is failed immediately because they are currently offline, the disable button way does not work.
2. I did explicitly specify mutation retry to `false`, however, this retry behavior is due to network connection recover:
```
If mutations fail because the device is offline, they will be retried in the same order when the device reconnects.
```

**Reply by TkDodo** [reply]
> If mutations fail because the device is offline, they will be retried in the same order when the device reconnects.

but this is only true if you have retries enabled. If you have no retries, nothing will be retried, because you have no retries :)
